src/node/full_node.cpp

```diff
@@ -142,6 +142,12 @@ void FullNode::init() {
           *jsonrpc_io_ctx_, boost::asio::ip::tcp::endpoint{conf_.rpc->address, *conf_.rpc->ws_port}, node_addr);
       jsonrpc_api_->addConnector(jsonrpc_ws_);
     }
+
+    if (conf_.rpc->gql_port) {
+      jsonrpc_http_ = make_shared<net::RpcServer>(
+          *jsonrpc_io_ctx_, boost::asio::ip::tcp::endpoint{conf_.rpc->address, *conf_.rpc->gql_port}, node_addr,
+          final_chain_, net::RpcServer::GraphQlType);
+    }
```

Comment:
- Is this ok ? Because you are overriding the value of jsonrpc_http_, which was already initiliazed a few lines above in case http_port was specified...
- did you consider also graphql support over websockets ?  
- I guess we want to support both jsonrpc and graphql at the same time ? If yes then we would have to have separate servers and not a single rpc server, which is either jsonrpc or graphql..
- it would be nice to differenciate these ports also in config, we could for example separate jsonrpc and graphql settings or if we keep them both under rpc, then we could have something like  rpc: { jsonrpc: {}, graphql: {} }
- config values validation might be added to (I already created such function) to make sure that people won't accidentaly specify same port for graphql and jsonrpc, http and ws, etc...

Reply:
No, I will fix that. I wanted to have two separate http server on separate ports. I do have already separate ports in the config

Reply:
This entire implementation of graphql is not very well tested. The API is maybe 90% implemented. Not 100%. There are some methods I was not sure how to implement. Also I am not sure if all the data formats I am returning are exactly correct. We need to start working with some clients. Either something existing that uses graphql or someone in Taraxa should start building something based on graphql and then we can fix stuff once they are being used.

Reply:
yeah sure, ports naming is just little inconsistent thats what I wanted to sa, if I see config like rpc : { http_port:<port>, ws_port:<port>, gql_port:<port> } it does not say much if gql port is related to http or ws... Something like jsonrpc : { http_port:<port>, ws_port:<port>}, graphql : { http_port:<port>, ws_port:<port>} would be more intuitive I think as it clearly separates jsonrpc and graphql settings

Reply:
I agree and I think that we should have a redo of everything in the configuration file. There is a lot of things inside the config file that could be better named and grouped but maybe it would be better to do that in a one commit/PR because each time we make individual changes to existing config properties it creates problems when we want to redeploy nodes to some existing specific configurations

Reply:
ok, @justinsnapp I guess we could put this config refactor task somewhere to baglog and look at it when we have more time...